Add release issue template

jrbourbeau opened the pull request:
Closes https://github.com/dask/community/issues/247

cc @fjetter @jakirkham @jsignell @jacobtomlinson

jakirkham commented on .github/ISSUE_TEMPLATE/release.md:
```diff
@@ -0,0 +1,31 @@
+| Release version | YYYY.MM.X |
+| Planned release date | YYYY-MM-DD |
+| Status | On-track |
+| Release manager | @github_handle |
```
Wonder if this is redundant for us given it is often the same person and very often the person who raised the issue

jrbourbeau replied:
Yeah, that's a good point that this will often be redundant. I have a slight preference for being explicit here and including this line. I don't think it's a big deal either way though

jakirkham replied:
It would be a little odd if someone assigned someone else as release manager here. That's the only reason this comes to mind. Though no strong feelings

jrbourbeau replied:
Yeah, agreed. I think explicitly naming the release manager here has a small amount of value for someone who is less familiar with our release process. If, for example, a maintainer of some project that uses Dask and has an upstream build identifies a large breaking change then they would see the GitHub handle for the person who is the release manager and could ping them directly. Again, probably not a huge deal either way

Let's go ahead and keep this line for now, try it out, and if we find it cumbersome then we can remove it

jakirkham commented on .github/ISSUE_TEMPLATE/release.md:
```diff
@@ -0,0 +1,31 @@
+- [ ] ...
+- [ ] ...
+
+## Comments
+
+...
```
What would go in this section?

jrbourbeau replied:
Just wanted to leave a place for jotting down anything else that's relevant. For example, I often cc you, Ben, Julia, and Florian and ask about any known blockers. That's one thing I anticipate going in this section

jakirkham replied:
Maybe it is worth having a stakeholders section?

jrbourbeau replied:
I'm personally okay with just a free-form comments section to start. Happy to change in the future if we think a dedicated stakeholders section would be useful for the group

jakirkham commented on .github/ISSUE_TEMPLATE/release.md:
```diff
@@ -0,0 +1,31 @@
+|  |  |
+| --- | --- |
+| Release version | YYYY.MM.X |
+| Planned release date | YYYY-MM-DD |
```
Do we need to worry about a timezone here (given it could be off +/-1 day)?

jrbourbeau replied:
I don't think timezone considerations will be will a big deal for this template (others should feel free to chime in if I'm missing something)

jakirkham replied:
Probably not enough to include in the date itself. Though maybe we could say something like this?

```suggestion
| Planned release date (US Central) | YYYY-MM-DD |
```

No strong feelings about what timezone goes there. Could see a case for US Eastern or similar.

jacobtomlinson replied:
Releases often happen after the end of my work day. But I generally expect that so not sure if there is any real value in being explicit here.

jakirkham reviewed (commented):
Thanks James! 🙏

One high level question that comes up when considering this is how we manage requests for releases (since these come up periodically) and separate them from release tracking?

Had a few questions on finer points below

jrbourbeau commented:
> One high level question that comes up when considering this is how we manage requests for releases (since these come up periodically) and separate them from release tracking?

That's a fair point. This doesn't happen that often, but you're right that it does come up from time to time. Maybe open a community issue still with the release request and then the release manager opens up a tracking issue? Do you have thoughts on a process here?

jakirkham commented:
Yeah that seems fine

We may also want to note in this template or in the issue selection GUI that this is intended for release managers only (and requests for a release should be a new issue)

jrbourbeau commented:
> We may also want to note in this template or in the issue selection GUI that this is intended for release managers only (and requests for a release should be a new issue)

I updated the `about` section of the template to mention release managers and added a comment to the template which says the template is explicitly for release managers and if someone wants to request a new release, they should open a different issue in the `dask/community` repo

jsignell reviewed (approved):
I'm fine with trying this out. We can always tweak it as we go.

jrbourbeau commented:
Thanks @jakirkham @jsignell for reviewing! 

> I'm fine with trying this out. We can always tweak it as we go.

+1
